### DISCUSSION TITLE
Welcome to rs_pbrt Discussions!

### DISCUSSION THREAD

**Discussion by wahn** [wahn/rs_pbrt#130] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by benjamin-godon** [comment]
Hello Jan and everybody here,

Beside being a professional C++/Python CG programmer it's my personal interest to study rs-pbrt.
Coming from the C++ world, I see Rust bringing a lot of hope for simpler/cleaner code.
Thank you so much for sharing your work as open source !!!

Benjamin

**Reply by wahn** [reply]
Hi Benjamin,

Thanks. I do believe that the Rust version is easier to compile etc. I have tried it on several Linux distros, MacOS, and Windows 10. With C++ you obviously have access to a lot of libraries being written in C/C++ over the years, while the Rust ecosystem is still growing, but you can mix C/C++ and Rust both ways. To get started with `rs-pbrt`:

- [web page](https://www.rs-pbrt.org/)
- [Read the Docs](https://rs-pbrt.readthedocs.io/en/latest/?badge=latest) ... I just started with that
- [about](https://www.rs-pbrt.org/about/) ... short text about the book, the C++ code, and the Rust version
- [Getting started](https://www.rs-pbrt.org/blog/getting-started/index.html) ... this will move finally into **Read the Docs**
- [release notes](https://www.rs-pbrt.org/tags/release/)
- [Rust code/crate documentation](https://www.rs-pbrt.org/doc/crates/pbrt/index.html) on **rs-pbrt.org** (updated only for releases)
- [Same as above](https://www.janwalter.org/doc/rust/pbrt/) on **janwalter.org** (updated sometimes inbetween releases)
- **rs-pbrt@posteo.de** for email questions

For the [Rust code/crate documentation](https://www.rs-pbrt.org/doc/crates/pbrt/index.html) you can click through the documentation. I would suggest starting with the [render loop](https://www.rs-pbrt.org/doc/crates/pbrt/core/integrator/enum.SamplerIntegrator.html#method.render) for all [sampler integrators](https://www.rs-pbrt.org/doc/crates/pbrt/core/integrator/enum.SamplerIntegrator.html) (ambient occlusion, direct lighting, whitted ray tracing, uni-directional path tracing, volumetric path tracing). The other algorithms (bi-directional path tracing, MLT, and Stochastic Progressive Photon Mapping) provide their own version of the render loop. From there you can click through related classes/structs/enums/traits and don't forget that there is an [online version](http://www.pbr-book.org/) of the book (obviously talking about the C++ version), but I made the Rust code very similar to the original.

**Reply by wahn** [reply]
And there is a [Discord server](https://discord.gg/ZEd2NFK45a) as well ...
